canvas.py

```diff
@@ -0,0 +1,72 @@
+from enum import Enum
+import cv2
+import numpy as np
+
+class Colour(Enum):
+    """ Predefined colours for use with Canvas """
+    # Remember, OpenCV uses BGR(A) not RGB(A)
+    white = (255, 255, 255, 255)
+    black = (0, 0, 0, 255)
+    transparentBlack = (0, 0, 0, 0)
+    blue = (255, 0, 0, 255)
+    green = (0, 255, 0, 255)
+    red = (0, 0, 255, 255)
+
+class Canvas():
+    """ A writeable image, for creating overlay content """
+
+    def __init__(self, width, height):
+        """ Initialises to plain black and transparent """
+        self.width = width
+        self.height = height
+        self.img = None
+        self.clear()
+
+    def clear(self):
+        """ Sets the entire canvas contents to transparent black """
+        self.img = np.zeros((self.height, self.width, 4), np.uint8)
+
+    @staticmethod
+    def _get_colour_tuple(colour):
+        """ Internal method - takes a 3-tulpe, 4-tuple or Colour class and
+            returns a 4-tuple colour """
+        if isinstance(colour, Colour):
```

Comment:
Duck typing but i'll let it go since you're supporting both `Colour` and 'rgb'

Reply:
I felt like it was okay because it adds to the usability of the canvas interface. Regardless, I've added some type hints which might make things clearer (or more complicated...)